python/strings_udf/cpp/include/cudf/strings/udf/split.cuh

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "udf_string.cuh"
+
+#include <cudf/strings/detail/split_utils.cuh>
+#include <cudf/strings/string_view.cuh>
+
+namespace cudf {
+namespace strings {
+namespace udf {
+
+/**
+ * @brief Split string using given string
+ *
+ * The caller must allocate an array of cudf::string_view to be filled
+ * in by this function. Calling this with a `nullptr` for the `result`
+ * will return the number of elements.
```

Comment:
This makes it sound like the returned value is different depending on whether or not `result == nullptr`. I would reword:
```suggestion
 * If `result != nullptr`, the split elements are copied into `result`.
```

Reply:
I disagree with this. I like the current wording.

Reply:
I added a separate `count_tokens` function which may help with the ambiguity.